common/v2/services/LocalCache/LocalCache.ts

```diff
@@ -81,91 +69,105 @@ export const initNodeOptions = () => {
-      if (newStorage.contractOptions[entry].network === en) {
+    const newAssets: string[] = [];
+    Object.keys(newStorage.contracts).map(entry => {
+      if (newStorage.contracts[entry].networkId === en) {
         newContracts.push(entry);
       }
     });
-    Object.keys(newStorage.assetOptions).map(entry => {
-      if (newStorage.assetOptions[entry].network === en) {
-        newAssetOptions.push(entry);
+    Object.keys(newStorage.assets).map(entry => {
+      if (newStorage.assets[entry].networkId === en) {
+        newAssets.push(entry);
       }
     });
-    const newLocalNetwork: types.NetworkOptions = {
+    const baseAssetID = utils.generateUUID();
+    const newLocalNetwork: types.Network = {
       contracts: newContracts,
-      assets: [STATIC_NETWORKS_INITIAL_STATE[en].id, ...newAssetOptions],
+      assets: [...newAssets],
       nodes: [],
+      baseAsset: baseAssetID,
       id: STATIC_NETWORKS_INITIAL_STATE[en].id,
       name: STATIC_NETWORKS_INITIAL_STATE[en].name,
-      unit: STATIC_NETWORKS_INITIAL_STATE[en].unit,
       chainId: STATIC_NETWORKS_INITIAL_STATE[en].chainId,
       isCustom: STATIC_NETWORKS_INITIAL_STATE[en].isCustom,
       color: STATIC_NETWORKS_INITIAL_STATE[en].color,
       blockExplorer: {},
       tokenExplorer: {},
-      tokens: {},
-      dPathFormats: STATIC_NETWORKS_INITIAL_STATE[en].dPathFormats,
+      dPaths: {
+        ...STATIC_NETWORKS_INITIAL_STATE[en].dPathFormats,
+        default: STATIC_NETWORKS_INITIAL_STATE[en].dPathFormats[InsecureWalletName.MNEMONIC_PHRASE]
+      },
       gasPriceSettings: STATIC_NETWORKS_INITIAL_STATE[en].gasPriceSettings,
       shouldEstimateGasPrice: STATIC_NETWORKS_INITIAL_STATE[en].shouldEstimateGasPrice
     };
-    const newLocalAssetOption: types.AssetOption = {
+    const newLocalAssetOption: types.Asset = {
+      uuid: baseAssetID,
       name: STATIC_NETWORKS_INITIAL_STATE[en].name,
-      network: en,
+      networkId: en,
       ticker: en,
       type: 'base',
       decimal: 18,
       contractAddress: null
     };
-    newStorage.networkOptions[en] = newLocalNetwork;
-    newStorage.assetOptions[STATIC_NETWORKS_INITIAL_STATE[en].id] = newLocalAssetOption;
+    newStorage.networks[en] = newLocalNetwork;
+    newStorage.assets[baseAssetID] = newLocalAssetOption;
   });
   setCache(newStorage);
 };
 
-export const initAssetOptions = () => {
+export const initAssets = () => {
   const newStorage = getCacheRaw();
-  const assets = AssetOptionsData();
+  const assets = AssetsData();
   Object.keys(assets).map(en => {
-    newStorage.assetOptions[en] = assets[en];
-    newStorage.networkOptions[assets[en].network].assets.push(en);
+    if (assets[en] && assets[en].networkId) {
+      const uuid = utils.generateUUID();
+      const networkName = assets[en].networkId;
+      assets[en].uuid = uuid;
+      newStorage.assets[uuid] = assets[en];
+      newStorage.networks[networkName!].assets.push(uuid);
```

Comment:
Is networkName always known to be defined at this point? If so, please explain why. If not, this should be wrapped in `if (networkName)` and `!` should be removed.

Reply:
Nope. Some assets aren't linked to networks (fiat assets). I've added the conditional locally.